### PR TITLE
CP-51325: Re-add and update patches

### PR DESCRIPTION
**Pull request by ElijahSadorra**
Whilst writing CoP testing, encountered a bug which needed these previous patches. Have now updated the patches to be compatible with the new version of the library.

### CODE REVIEW

**Review by CitrixChris — APPROVED** [review]

File: patches/patch-sharpziplib-CA63358-TarBuffer.patch [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+# SUCH DAMAGE.
+
+--- a/SharpZipLib-1.3.3/src/ICSharpCode.SharpZipLib/Tar/TarBuffer.cs	2011-06-17 23:09:32.000000000 +0100
++++ b/SharpZipLib-1.3.3/src/ICSharpCode.SharpZipLib/Tar/TarBuffer.cs	2011-08-01 20:14:18.000000000 +0100
+@@ -286,55 +286,6 @@ namespace ICSharpCode.SharpZipLib.Tar
```
Comment (kc284) [inline patches/patch-sharpziplib-CA63358-TarBuffer.patch:32]: How did we end up with code lines in the chunk headers? (Several more instances of it below).

File: patches/patch-sharpziplib-CA63358-TarBuffer.patch [hunk 2]
```diff
@@ -0,0 +1,126 @@
+# Copyright (c) Cloud Software Group, Inc.
```
Comment (kc284) [inline patches/patch-sharpziplib-CA63358-TarBuffer.patch:1]: Minor: I suggest keeping the exact name of the original files (i.e. `patch-sharpziplib-ca-63358-TarBuffer.diff`) so it's easier to diff them.

**Review by kc284 — CHANGES_REQUESTED** [review]

File: patches/patch-sharpziplib-CA63358-TarBuffer.patch (hunk 1 above)
Comment (ElijahSadorra) [reply patches/patch-sharpziplib-CA63358-TarBuffer.patch:32]: I'm assuming you mean the "namespace ICSharpCode.SharpZipLib.Tar"? If that's the case, it was auto generated by the git diff command. I tested with and without "namespace ICSharpCode.SharpZipLib.Tar", both are identical so happy to keep it so it is similar for future patches and also happy to remove it to keep the old convention.

Comment (kc284) [reply patches/patch-sharpziplib-CA63358-TarBuffer.patch:32]: As long as the patch applies successfully it doesn't matter too much, but I think it is cleaner without it.

Comment (ElijahSadorra) [reply patches/patch-sharpziplib-CA63358-TarBuffer.patch:32]: Ok cool, yes both apply cleanly so will just remove it.

**Comment by ElijahSadorra** [comment]
Changed name and removed unnecessary use of namespace in the patches. When applying patches using `./build`, all patches was applied cleanly and produced the same `.dll`.

**Review by kc284 — APPROVED** [review]

**Pull request merged**